Doc/library/sqlite3.rst

```diff
@@ -601,25 +601,15 @@ Connection objects
 
       Example:
 
-      .. testcode::
-
-         def dict_factory(cursor, row):
-             d = {}
-             for idx, col in enumerate(cursor.description):
-                 d[col[0]] = row[idx]
-             return d
-
-         con = sqlite3.connect(":memory:")
-         con.row_factory = dict_factory
-         cur = con.execute("SELECT 1 AS a")
-         print(cur.fetchone()["a"])
-
-         con.close()
-
-      .. testoutput::
-         :hide:
+      .. doctest::
 
-         1
+         >>> def dict_factory(cursor, row):
+         ...     return {col[0]: row[i] for i, col in enumerate(cursor.description)}
```

Comment:
How does `col` look like?  What about `cursor.description`?  Maybe add a comment before the return to explain its content and/or why we use `col[0]`?

Also wouldn't something like this be better/clearer:
```suggestion
         ...     col_names = [col[0] for col in cursor.description]
         ...     return {col_name: value for col_name, value in zip(col_names, row)}
```
This will also allow you to do something like `[name for (name, type, ...) in cursor.description]` instead of having to explain the content/structure of cursor.description.

Reply:
Yeah, that's even better. Thanks!

Reply:
```suggestion
         ...     col_names = [col[0] for col in cursor.description]
         ...     return {key: value for key, value in zip(col_names, row)}
```